boututils/calculus.py

```diff
@@ -44,18 +44,20 @@ def deriv(*args, **kwargs):
 
     n = var.size
     if periodic:
+        # assume uniform grid 
+        dx = x[1] - x[0]
         # Use FFTs to take derivatives
         f = rfft(var)
         f[0] = 0.0  # Zero constant term
         if n % 2 == 0:
             # Even n
             for i in arange(1, old_div(n, 2)):
```

Comment:
```suggestion
            for i in range(1, (n+1) // 2):
```
In that case the `else` branch can be skipped, and only setting `f[-1]=0.0` needs to be guarded by an if.

Reply:
Yeah, xbout could be enhanced with derivative evaluation using DFT. At the moment it uses a 2nd order central finite difference method